lib/uffizzi/clients/api/api_client.rb

```diff
@@ -19,15 +19,49 @@ def fetch_projects(hostname)
     build_response(response)
   end
 
+  def add_compose_file(hostname, params, project_slug)
+    uri = compose_file_uri(hostname, project_slug)
+    response = Uffizzi::HttpClient.make_request(uri, :post, true, params)
+
+    build_response(response)
+  end
+
+  def remove_compose_file(hostname, params, project_slug)
+    uri = compose_file_uri(hostname, project_slug)
+    response = Uffizzi::HttpClient.make_request(uri, :delete, true, params)
+
+    build_response(response)
+  end
+
+  def describe_compose_file(hostname, params, project_slug)
+    uri = compose_file_uri(hostname, project_slug)
+    response = Uffizzi::HttpClient.make_request(uri, :get, true, params)
+
+    build_response(response)
+  end
+
+  def validate_compose_file(hostname, params, project_slug)
+    uri = validate_compose_file_uri(hostname, project_slug)
+    response = Uffizzi::HttpClient.make_request(uri, :get, true, params)
+
+    build_response(response)
+  end
```

Comment:
We need to think about how to not make this basic API client too overloaded with methods